checkbox-support/checkbox_support/scripts/image_checker.py

```diff
@@ -7,16 +7,42 @@
 import argparse
 from os.path import exists
 from checkbox_support.snap_utils.system import on_ubuntucore
+import shutil
+from subprocess import PIPE, run
 
 
-def get_type():
+def get_type() -> str:
     """
     Return the type of image.
     """
     return "core" if on_ubuntucore() else "classic"
 
 
-def get_source():
+def has_desktop_environment() -> bool:
+    """
+    Returns whether there's a desktop environment
+    """
+    if not shutil.which("dpkg") or on_ubuntucore():
+        # core doesn't have dpkg
+        return False
```

Comment:
```suggestion
    if on_ubuntucore():
        # Ubuntu Core doesn't have a desktop environment by default. 
        return False
```

I think `dpkg` is the default command installed in Ubuntu. Therefore, I think we don't have to check it.